File: zh-Hant/codes/swift/chapter_tree/binary_tree.swift
/**
 * File: binary_tree.swift
 * Created Time: 2023-01-18
 * Author: nuomi1 (nuomi1@qq.com)
 */

import utils

@main
enum BinaryTree {
    /* Driver Code */
    static func main() {
        /* 初始化二叉树 */
        // 初始化节点
        let n1 = TreeNode(x: 1)
        let n2 = TreeNode(x: 2)
        let n3 = TreeNode(x: 3)
        let n4 = TreeNode(x: 4)
        let n5 = TreeNode(x: 5)
        // 构建节点之间的引用（指针）
        n1.left = n2
        n1.right = n3
        n2.left = n4
        n2.right = n5
        print("\n初始化二叉树\n")
        PrintUtil.printTree(root: n1)

        /* 插入与删除节点 */
        let P = TreeNode(x: 0)
        // 在 n1 -> n2 中间插入节点 P
        n1.left = P
        P.left = n2
        print("\n插入节点 P 后\n")
        PrintUtil.printTree(root: n1)
        // 删除节点 P
        n1.left = n2
        print("\n删除节点 P 后\n")
        PrintUtil.printTree(root: n1)
    }
}

File: zh-hant/codes/java/chapter_tree/binary_tree.java
/**
 * File: binary_tree.java
 * Created Time: 2022-11-25
 * Author: krahets (krahets@163.com)
 */

package chapter_tree;

import utils.*;

public class binary_tree {
    public static void main(String[] args) {
        /* 初始化二元樹 */
        // 初始化節點
        TreeNode n1 = new TreeNode(1);
        TreeNode n2 = new TreeNode(2);
        TreeNode n3 = new TreeNode(3);
        TreeNode n4 = new TreeNode(4);
        TreeNode n5 = new TreeNode(5);
        // 構建節點之間的引用（指標）
        n1.left = n2;
        n1.right = n3;
        n2.left = n4;
        n2.right = n5;
        System.out.println("\n初始化二元樹\n");
        PrintUtil.printTree(n1);

        /* 插入與刪除節點 */
        TreeNode P = new TreeNode(0);
        // 在 n1 -> n2 中間插入節點 P
        n1.left = P;
        P.left = n2;
        System.out.println("\n插入節點 P 後\n");
        PrintUtil.printTree(n1);
        // 刪除節點 P
        n1.left = n2;
        System.out.println("\n刪除節點 P 後\n");
        PrintUtil.printTree(n1);
    }
}
